contracts/marketplace/src/execute.rs

```diff
@@ -233,7 +233,7 @@ pub fn execute_set_ask(
 
     nonpayable(&info)?;
     price_validate(deps.storage, &price)?;
-    only_owner(deps.as_ref(), &info, collection.clone(), token_id)?;
+    only_owner(deps.as_ref(), &info, &collection, token_id)?;
```

Comment:
is there a memory usage gain by passing ref?

Reply:
Hopefully someday indexed maps will work with `&Addr` keys, but right now they only work with `Addr`.